### ISSUE TITLE
Tiles wrongly placed after switching view

### ISSUE THREAD

**Issue by Enkosz** [maplibre/maplibre-gl-leaflet#49]
Hello there, 

I've been trying to use `maplibre-gl` with `react-leaflet` to display satellite and roadmap layers from an external provider. I noticed a random behaviour when switching from one to another. 

As you can notice from the screenshot below, when switching from satellite to roadmap the map / layers are shifted to the left and not correctly loading. Interacting with the map however will make it work again.

<img width="1250" alt="Screenshot 2023-09-05 at 13 35 10" src="https://github.com/maplibre/maplibre-gl-leaflet/assets/35410110/cf9cb3a2-8f39-468b-861f-3564e8334b47">
<img width="1250" alt="Screenshot 2023-09-05 at 13 37 48" src="https://github.com/maplibre/maplibre-gl-leaflet/assets/35410110/6c07eadb-1f3f-4e76-9b8d-4e5cba34e8e5">

React Leaflet tile layer:
```import { createTileLayerComponent, updateGridLayer, withPane } from '@react-leaflet/core';
import L from 'leaflet';

import '@maplibre/maplibre-gl-leaflet';

const MapLibreTileLayer = createTileLayerComponent(
    function createTileLayer({ url, attribution, ...options }, context) {
        const instance = L.maplibreGL(
            { style: url, attribution: attribution, noWrap: true },
            withPane(options, context),
        );
        return Object.freeze({ instance, context });
    },
    function updateTileLayer(layer, props, prevProps) {
        updateGridLayer(layer, props, prevProps);

        const { url, attribution } = props;
        if (url != null && url !== prevProps.url) {
            layer.getMaplibreMap().setStyle(url);
        }

        if (attribution != null && attribution !== prevProps.attribution) {
            layer.options.attribution = attribution;
        }
    },
);

export default MapLibreTileLayer;
```

versions:
```
@maplibre/maplibre-gl-leaflet": "0.0.19",
react-leaflet: "3.2.5",
maplibre-gl: "3.3.1",
leaflet: "1.9.3",
```

**Comment by lseelenbinder** [comment]
@Enkosz Have you tried calling `layer.getMaplibreMap().resize()` or `layer.getMaplibreMap().redraw()`?

I suspect it's not properly triggering a redraw after `setStyle` is being called.

**Comment by Enkosz** [comment]
> @Enkosz Have you tried calling `layer.getMaplibreMap().resize()` or `layer.getMaplibreMap().redraw()`?
> 
> I suspect it's not properly triggering a redraw after `setStyle` is being called.

Yep, I tried both in the Tile component and also some more "heavy" approaches like invalidating the Leaflet map instance when switching to another Layer. I couldn't find the reason why it happens randomly and not for every instance. 
Might having a reproducible example help?

**Comment by lseelenbinder** [comment]
Yes, if you can create a reproducer, that'd be great.

**Comment by Enkosz** [comment]
@lseelenbinder Hey hey, got any news?

**Comment by mapcentia** [comment]
I'm having the same problem. Did you find a solution?
